src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -397,6 +397,28 @@ internal void SetTypeWithAnnotations(TypeWithAnnotations newType)
             }
         }
 
+        /// <summary>
+        /// Gets a new local symbol with the given TypeWithAnnotations as the new type. This
+        /// type should be identical to the original except for nullability.
+        /// </summary>
+        internal LocalSymbol WithAnalyzedType(TypeWithAnnotations analyzedType)
+        {
+            Debug.Assert(analyzedType.Equals(TypeWithAnnotations, TypeCompareKind.AllNullableIgnoreOptions | TypeCompareKind.IgnoreTupleNames));
```

Comment:
>TypeCompareKind.IgnoreTupleNames [](start = 109, length = 32)

Are there cases where the analyzed type differs from the original by tuple names? Is that a bug?

Reply:
Yes, tracked by [https://github.com/dotnet/roslyn/issues/34993](https://github.com/dotnet/roslyn/issues/34993).

---
In reply to: [306385539](https://github.com/dotnet/roslyn/pull/37322#discussion_r306385539) [](ancestors = 306385539)